File: modules/swagger-core/src/test/scala/GuavaTest.scala
import com.fasterxml.jackson.datatype.guava.GuavaModule
import io.swagger.converter.ModelConverters
import io.swagger.util.Json
import matchers.SerializationMatchers.serializeToJson
import models.GuavaModel
import org.junit.runner.RunWith
import org.scalatest.{FlatSpec, Matchers}
import org.scalatest.junit.JUnitRunner

@RunWith(classOf[JUnitRunner])
class GuavaTest extends FlatSpec with Matchers {
  val m = Json.mapper()

  it should "convert a model with Guava optionals" in {
    m.registerModule(new GuavaModule())

    val schemas = ModelConverters.getInstance().read(classOf[GuavaModel])
    schemas should serializeToJson( """{"GuavaModel":{"type": "object","properties":{"name":{"type":"string"}}}}""")
  }
}

File: modules/swagger-guava-test/src/test/java/io/swagger/GuavaTest.java
package io.swagger;

import com.fasterxml.jackson.databind.node.ObjectNode;
import io.swagger.converter.ModelConverters;
import io.swagger.models.GuavaModel;
import io.swagger.models.Model;
import io.swagger.util.Json;
import org.testng.annotations.Test;

import java.io.IOException;
import java.util.Map;

import static org.testng.Assert.assertEquals;

public class GuavaTest {

    @Test(description = "convert a model with Guava optionals")
    public void convertModelWithGuavaOptionals() throws IOException {
        final Map<String, Model> schemas = ModelConverters.getInstance().read(GuavaModel.class);
        assertEquals(
            Json.mapper().convertValue(schemas, ObjectNode.class),
            Json.mapper().readValue(
                "{\n" +
                "    \"GuavaModel\": {\n" +
                "        \"type\": \"object\",\n" +
                "        \"properties\": {\n" +
                "            \"name\": {\n" +
                "                \"type\": \"string\"\n" +
                "            }\n" +
                "        }\n" +
                "    }\n" +
                "}\n", ObjectNode.class));
    }
}
